lib/active_merchant/billing/gateways/trexle.rb

```diff
@@ -0,0 +1,218 @@
+      def purchase(money, creditcard, options = {})
+        post = {}
+
+        add_amount(post, money, options)
+        add_customer_data(post, options)
+        add_invoice(post, options)
+        add_creditcard(post, creditcard)
+        add_address(post, creditcard, options)
+        add_capture(post, options)
+
+        commit(:post, 'charges', post, options)
+      end
+
+      # Create a customer and associated credit card. The token that is returned
+      # can be used instead of a credit card parameter in the purchase method
+      def store(creditcard, options = {})
+        post = {}
+
+        add_creditcard(post, creditcard)
+        add_customer_data(post, options)
+        add_address(post, creditcard, options)
+        commit(:post, 'customers', post, options)
+      end
+
+      # Refund a transaction
+      def refund(money, token, options = {})
+        commit(:post, "charges/#{CGI.escape(token)}/refunds", { :amount => amount(money) }, options)
+      end
+
+      # Authorize an amount on a credit card. Once authorized, you can later
+      # capture this charge using the charge token that is returned.
+      def authorize(money, creditcard, options = {})
+        options[:capture] = false
+
+        purchase(money, creditcard, options)
+      end
+
+      # Captures a previously authorized charge. Capturing only part of the original
+      # authorization is currently not supported.
+      def capture(money, token, options = {})
+        commit(:put, "charges/#{CGI.escape(token)}/capture", { :amount => amount(money) }, options)
+      end
+
+      # Updates the credit card for the customer.
+      def update(token, creditcard, options = {})
+        post = {}
+
+        add_creditcard(post, creditcard)
+        add_customer_data(post, options)
+        add_address(post, creditcard, options)
+        commit(:put, "customers/#{CGI.escape(token)}", post, options)
+      end
+
+      def supports_scrubbing
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
+          gsub(/(number\\?":\\?")(\d*)/, '\1[FILTERED]').
+          gsub(/(cvc\\?":\\?")(\d*)/, '\1[FILTERED]')
+      end
+      private
+
+      def add_amount(post, money, options)
+        post[:amount] = amount(money)
+        post[:currency] = (options[:currency] || currency(money))
+        post[:currency] = post[:currency].upcase if post[:currency]
+      end
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email] if options[:email]
+        post[:ip_address] = options[:ip] if options[:ip]
+      end
+
+      def add_address(post, creditcard, options)
+        return if creditcard.kind_of?(String)
+        address = (options[:billing_address] || options[:address])
+        return unless address
+
+        post[:card] ||= {}
+        post[:card].merge!(
+          :address_line1 => address[:address1],
```

Comment:
According to the docs, it looks like `:address_line2` is supported, so we should have it here too.